src/Gaffer/Animation.cpp

```diff
@@ -55,6 +56,29 @@ class Animation::Interpolator : public IECore::RefCounted
 
 		IE_CORE_DECLAREMEMBERPTR( Interpolator )
 
+		enum class Hint
+		{
```

Comment:
I think everywhere else we use bitmasks, we declare the enum values as 1, 2, 4, 8 rather than bit indexes.  Using bit indexes with a `test( hint )` function works fine, don't think I've really got a preference either way ( maybe this is a better approach than what we're doing elsewhere ), but wondering if there is value in consistency.

Reply:
Yep, in the public API we've settled on power-of-two enum values rather than bit indexes, and eschewed the additional type safety of things like the `Hints` class in favour of reduction in boilerplate and additional class types. See `Plug::Flags` or `PathMatcher::Result` for examples. If we want to revisit I imagine we would use something like QFlags, so one template type can be used everywhere.

Likewise, we don't use separate factory classes. Instead we just have a couple of static methods on the class itself - for example, see `View::registerView()` and `View::create()`. Again, this reduces boilerplate and API surface area.

Since this is all private, I'm not too concerned either way, but my preference would be to simplify in line with the public API. It seems a little odd to take a more heavyweight approach privately than we do publicly.

Reply:
code now uses unsigned bitmasks and static methods instead of a factory class.